### DISCUSSION TITLE
Can't open web when deploy with docker compose

### DISCUSSION THREAD

**Discussion by gkl1368** [firefly-iii/firefly-iii#7115] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely me, not you](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

After update to 6.0 with docker compose, the web can't opened. I used the same config with v5. v5 worked ok. I have tried to clear the data in database, it can't save.

### Expected behaviour

Can't open the web

### Debug information

No

**Comment by JC5** [comment]
I'll need a bit more detail than this I'm afraid. Are there errors in your log files?

**Reply by gkl1368** [reply]
No errors, but I can paste the docker compose logs
FireflyIII  | Now in entrypoint.sh for Firefly III
FireflyIII  | Script:       1.0.21 (2022-02-17)
FireflyIII  | User:         'root'
FireflyIII  | Group:        'root'
FireflyIII  | Working dir:  '/var/www/html'
FireflyIII  | Build number:  407
FireflyIII  | Build date:    02-03-2023 21:24:37 CET
FireflyIII  | Now parsing _FILE variables.
FireflyIII  | done!
FireflyIII  | Touch DB file (if SQLite)...
FireflyIII  | Dump auto load...
FireflyIII  | Generating optimized autoload files
FireflyIII  | > Illuminate\Foundation\ComposerScripts::postAutoloadDump
FireflyIII  | Generated optimized autoload files containing 6250 classes
FireflyIII  | Discover packages...
FireflyIII  | 
FireflyIII  |    INFO  Discovering packages.  
FireflyIII  | 
FireflyIII  |   diglactic/laravel-breadcrumbs ......................................... DONE
FireflyIII  |   jc5/google2fa-laravel ................................................. DONE
FireflyIII  |   laravel/passport ...................................................... DONE
FireflyIII  |   laravel/sanctum ....................................................... DONE
FireflyIII  |   laravel/slack-notification-channel .................................... DONE
FireflyIII  |   laravel/ui ............................................................ DONE
FireflyIII  |   laravelcollective/html ................................................ DONE
FireflyIII  |   nesbot/carbon ......................................................... DONE
FireflyIII  |   nunomaduro/collision .................................................. DONE
FireflyIII  |   nunomaduro/termwind ................................................... DONE
FireflyIII  |   rcrowe/twigbridge ..................................................... DONE
FireflyIII  |   spatie/laravel-ignition ............................................... DONE
FireflyIII  | 
FireflyIII  | Current working dir is '/var/www/html'
FireflyIII  | Wait for the database.
FireflyIII  | wait-for-it.sh: waiting 60 seconds for fireflyiii_db:3306
FireflyIII  | wait-for-it.sh: fireflyiii_db:3306 is available after 0 seconds
FireflyIII  | DB is up.
FireflyIII  | Wait another 5 seconds in case the DB needs to boot.
FireflyIII  | Done waiting for the DB to boot.
FireflyIII  | Will not build the locales...
FireflyIII  | Run various artisan commands...
FireflyIII  | Running migration commands...
FireflyIII  | DSN is mysql:host=fireflyiii_db;port=3306;charset=utf8mb4
FireflyIII  | Database "fireflyiii" exists.
FireflyIII  | 
FireflyIII  |    INFO  Nothing to migrate.  
FireflyIII  | 
FireflyIII  |    INFO  Seeding database.  
FireflyIII  | 
FireflyIII  |   Database\Seeders\AccountTypeSeeder ................................. RUNNING  
FireflyIII  |   Database\Seeders\AccountTypeSeeder ........................... 27.44 ms DONE  
FireflyIII  | 
FireflyIII  |   Database\Seeders\TransactionCurrencySeeder ......................... RUNNING  
FireflyIII  |   Database\Seeders\TransactionCurrencySeeder ................... 14.12 ms DONE  
FireflyIII  | 
FireflyIII  |   Database\Seeders\TransactionTypeSeeder ............................. RUNNING  
FireflyIII  |   Database\Seeders\TransactionTypeSeeder ....................... 21.64 ms DONE  
FireflyIII  | 
FireflyIII  |   Database\Seeders\PermissionSeeder .................................. RUNNING  
FireflyIII  |   Database\Seeders\PermissionSeeder ............................. 1.63 ms DONE  
FireflyIII  | 
FireflyIII  |   Database\Seeders\LinkTypeSeeder .................................... RUNNING  
FireflyIII  |   Database\Seeders\LinkTypeSeeder ............................... 3.31 ms DONE  
FireflyIII  | 
FireflyIII  |   Database\Seeders\ConfigSeeder ...................................... RUNNING  
FireflyIII  | [2023-03-04 15:20:00] production.WARNING: Database entry exists. Update to latest version (19)  
FireflyIII  |   Database\Seeders\ConfigSeeder ................................ 21.75 ms DONE  
FireflyIII  | 
FireflyIII  |   Database\Seeders\UserRoleSeeder .................................... RUNNING  
FireflyIII  |   Database\Seeders\UserRoleSeeder .............................. 25.67 ms DONE  
FireflyIII  | 
FireflyIII  |   Database\Seeders\ExchangeRateSeeder ................................ RUNNING  
FireflyIII  |   Database\Seeders\ExchangeRateSeeder .......................... 16.32 ms DONE  
FireflyIII  | 
FireflyIII  | Going to decrypt the database.
FireflyIII  | No decryption required for table "accounts".
FireflyIII  | No decryption required for table "attachments".
FireflyIII  | No decryption required for table "bills".
FireflyIII  | No decryption required for table "budgets".
FireflyIII  | No decryption required for table "categories".
FireflyIII  | No decryption required for table "piggy_banks".
FireflyIII  | No decryption required for table "preferences".
FireflyIII  | No decryption required for table "tags".
FireflyIII  | No decryption required for table "transaction_journals".
FireflyIII  | No decryption required for table "transactions".
FireflyIII  | No decryption required for table "journal_links".
FireflyIII  | Done!
FireflyIII  | Current working dir is '/var/www/html'
FireflyIII  | Command executed successfully.
FireflyIII  | Running upgrade commands...
FireflyIII  | This command has already been executed.
FireflyIII  | Database already seems to be migrated.
FireflyIII  | This command has already been executed.
FireflyIII  | This command has already been executed.
FireflyIII  | This command has already been executed.
FireflyIII  | This command has already been executed.
FireflyIII  | This command has already been executed.
FireflyIII  | This command has already been executed.
FireflyIII  | This command has already been executed.
FireflyIII  | No incorrectly stored credit card liabilities.
FireflyIII  | This command has already been executed.
FireflyIII  | This command has already been executed.
FireflyIII  | This command has already been executed.
FireflyIII  | This command has already been executed.
FireflyIII  | This command has already been executed.
FireflyIII  | This command has already been executed.
FireflyIII  | This command has already been executed.
FireflyIII  | Running verification commands...
FireflyIII  | All piggy bank events are correct.
FireflyIII  | Verified the content of 0 piggy bank events in 0.03 seconds.
FireflyIII  | All link types OK!
FireflyIII  | Verified link types in 0.01 seconds
FireflyIII  | All access tokens OK!
FireflyIII  | Verify access tokens in 0.01 seconds.
FireflyIII  | All transaction journals have correct bill information.
FireflyIII  | Verified bills / journals in 0.04 seconds
FireflyIII  | All budget limits are OK.
FireflyIII  | 0 different currencies are currently in use.
FireflyIII  | All currencies are correctly enabled or disabled.
FireflyIII  | Verified currencies in 0.08 seconds.
FireflyIII  | No invalid budget/journal entries.
FireflyIII  | Verified budget/journals in 0.02 seconds.
FireflyIII  | Amount integrity OK!
FireflyIII  | Verified amount integrity in 0.01 seconds
FireflyIII  | No zero-amount transaction journals.
FireflyIII  | Verified zero-amount integrity in 0.01 seconds
FireflyIII  | No orphaned journals.
FireflyIII  | No orphaned transactions.
FireflyIII  | No orphaned accounts.
FireflyIII  | Verified orphans in 0.04 seconds
FireflyIII  | No uneven transaction journals.
FireflyIII  | No empty transaction journals.
FireflyIII  | Verified empty journals in 0 seconds
FireflyIII  | Verified empty groups in 0.02 seconds
FireflyIII  | All account types are OK!
FireflyIII  | Verifying account types took 0.02 seconds
FireflyIII  | Verifying account order took 0.01 seconds
FireflyIII  | All meta fields are correct.
FireflyIII  | Renamed meta fields in 0.01 seconds
FireflyIII  | Going to verify 0 opening balance transactions.
FireflyIII  | There was nothing to fix in the opening balance transactions.
FireflyIII  | Verified all transaction group and journal title lengths.
FireflyIII  | Took 0.01 seconds.
FireflyIII  | Corrected recurring transactions in 0.04 seconds.
FireflyIII  | Updated inconsistent transaction groups.
FireflyIII  | All transaction journals are of the correct transaction type (in 0.01 seconds).
FireflyIII  | Verifying account preferences took 0.01 seconds
FireflyIII  | Validated group memberships in 0.25 seconds.
FireflyIII  | Running report commands...
FireflyIII  | Report on empty objects finished in 0.03 seconds
FireflyIII  | Report on total sum finished in 0.01 seconds
FireflyIII  | Restored OAuth keys from database.
FireflyIII  | Generating new OAuth keys...
FireflyIII  | Encryption keys already exist. Use the --force option to overwrite them.
FireflyIII  | Personal access client created successfully.
FireflyIII  | Client ID: 151
FireflyIII  | Client secret: ositK7YxhUB37FQthHMNOz4U7U05Un1P1MYtSAI3
FireflyIII  | Password grant client created successfully.
FireflyIII  | Client ID: 152
FireflyIII  | Client secret: L5koNR4iYWw0rtAapxN9AsD4bORm33Y09WKKUOMj
FireflyIII  | Updated version.
FireflyIII  | +------------------------------------------------------------------------------+
FireflyIII  | |                                                                              |
FireflyIII  | | Thank you for installing Firefly III, v6.0.0!                                |
FireflyIII  | |                                                                              |
FireflyIII  | |                                                                              |
FireflyIII  | +------------------------------------------------------------------------------+
FireflyIII  | You can safely ignore the error about the 'fully qualified domain name'.
FireflyIII  | [Sat Mar 04 15:20:12.351089 2023] [:crit] [pid 1] (38)Function not implemented: AH00141: Could not initialize random number generator
FireflyIII  | Now in entrypoint.sh for Firefly III
FireflyIII  | Script:       1.0.21 (2022-02-17)
FireflyIII  | User:         'root'
FireflyIII  | Group:        'root'
FireflyIII  | Working dir:  '/var/www/html'
FireflyIII  | Build number:  407
FireflyIII  | Build date:    02-03-2023 21:24:37 CET
FireflyIII  | Now parsing _FILE variables.
FireflyIII  | done!
FireflyIII  | Touch DB file (if SQLite)...
FireflyIII  | Dump auto load...
FireflyIII  | Generating optimized autoload files
FireflyIII  | > Illuminate\Foundation\ComposerScripts::postAutoloadDump
FireflyIII  | Generated optimized autoload files containing 6250 classes
FireflyIII  | Discover packages...
FireflyIII  | 
FireflyIII  |    INFO  Discovering packages.  
FireflyIII  | 
FireflyIII  |   diglactic/laravel-breadcrumbs ......................................... DONE
FireflyIII  |   jc5/google2fa-laravel ................................................. DONE
FireflyIII  |   laravel/passport ...................................................... DONE
FireflyIII  |   laravel/sanctum ....................................................... DONE
FireflyIII  |   laravel/slack-notification-channel .................................... DONE
FireflyIII  |   laravel/ui ............................................................ DONE
FireflyIII  |   laravelcollective/html ................................................ DONE
FireflyIII  |   nesbot/carbon ......................................................... DONE
FireflyIII  |   nunomaduro/collision .................................................. DONE
FireflyIII  |   nunomaduro/termwind ................................................... DONE
FireflyIII  |   rcrowe/twigbridge ..................................................... DONE
FireflyIII  |   spatie/laravel-ignition ............................................... DONE
FireflyIII  | 
FireflyIII  | Current working dir is '/var/www/html'
FireflyIII  | Wait for the database.
FireflyIII  | wait-for-it.sh: waiting 60 seconds for fireflyiii_db:3306
FireflyIII  | wait-for-it.sh: fireflyiii_db:3306 is available after 0 seconds
FireflyIII  | DB is up.
FireflyIII  | Wait another 5 seconds in case the DB needs to boot.

It seems it always restart in the docker

**Reply by JC5** [reply]
Ah. Your kernel is too old. Running on a Synology perhaps?

**Reply by gkl1368** [reply]
yes, In  synology  system

**Reply by JC5** [reply]
Sorry, there's nothing I can do for you.

**Reply by gkl1368** [reply]
ok. I will transfer this docker to other system with new kernel.

**Reply by 8bitgentleman** [reply]
I've also run into this issue running a docker container on a synology NAS. I was upgrading to firefly v6, is it possible do downgrade back after hitting this error?

**Comment by Ave-Rechts** [comment]
@JC5 

By no means I want to hijack this discussion, but I have the same issue and my logs look exactly the same.
So if it is preferable that I open a new discussion, let me know and I will do that.


Is this how u noticed that the kernel is outdated or something?
FireflyIII | [Sat Mar 04 15:20:12.351089 2023] [:crit] [pid 1] (38)Function not implemented: AH00141: Could not initialize random number generator

**Reply by JC5** [reply]
You have the same issue and it's not something I can help unfortunately.

**Reply by Ave-Rechts** [reply]
I can understand that. I am just trying to understand how u detected it and what is the oldest kernel version that is supported by 6.0 :)

**Reply by andrej-g** [reply]
If you look up `AH00141: Could not initialize random number generator` you can find a bunch of similar issues across many projects; e.g. [NextCloud](https://github.com/nextcloud/docker/issues/1574) or [WordPress](https://github.com/docker-library/wordpress/issues/630).

According to those two links the kernel should be 3.16 or newer. [More context](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=978045).

**Reply by Ave-Rechts** [reply]
That is exactly what I did, but I wasnt sure if it was the random number generator that caused 6.0 to break on my Synology, hence my question :) 
3.10.108 is what I have, sad life :(

**Reply by Chicane90** [reply]
Im facing the same Issue, we know that this breaks with Version 6.0 ?
So to "mitigate" this Issue, we could run Version 5.7.18
